6_temp_coop_fetch/src/task_fxns.R

```diff
@@ -0,0 +1,41 @@
+create_coop_taskplan <- function(wants) {
+
+  coop_fetch_step1 <- scipiper::create_task_step(
+    step_name = 'check_and_write_ind',
+    target_name = function(task_name, ...) {
+      scipiper::as_ind_file(file.path('6_temp_coop_fetch/in', task_name))
+    },
+    command = "gd_confirm_posted(ind_file = target_name)"
+  )
+
+  coop_fetch_step2 <- scipiper::create_task_step(
+    step_name = 'download',
+    target_name = function(task_name, ...) {
+      file.path('6_temp_coop_fetch/in', task_name)
+    },
+    command = function(target_name, ...) {
+      sprintf("gd_get('%s')", scipiper::as_ind_file(target_name))
+    }
+  )
+
+  task_plan <- scipiper::create_task_plan(
+    task_names = wants,
+    task_steps = list(coop_fetch_step1, coop_fetch_step2),
+    final_steps = 'download',
+    add_complete = FALSE
+  )
+
+  return(task_plan)
+}
+
+create_coop_fetch_makefile <- function(target_name, taskplan) {
+  create_task_makefile(
+    makefile = target_name,
+    task_plan = taskplan,
+    include = '6_temp_coop_fetch.yml',
+    packages = c('scipiper'),
+    file_extensions = c("ind"),
+    ind_dir = '6_temp_coop_fetch/log',
+    ind_complete = TRUE
```

Comment:
Cool - I was wondering about the same thing. In my parent remake file, I was using an object as the target for `loop_tasks` instead of the `.ind` file. I wasn't sure how I could program the exact indicator file target in w/o going into the logic of how the task table was built. For example, mine looks like this within the task_remake_file:

```yaml
  # --- Overall job --- #

  6_fetch_new_feather_tasks:
    command: sc_indicate(I('6_drivers/log/feather/6_fetch_new_feather_tasks.ind'), hash_depends=I(TRUE), depends_target=I('6_fetch_new_feather_tasks'), depends_makefile=I('6_fetch_new_feather_tasks.yml'))
    depends: 
      - 6_drivers/log/feather/NLDAS_time[0.24999]_x[221]_y[132]_var[dlwrfsfc].feather.ind
      - 6_drivers/log/feather/NLDAS_time[0.24999]_x[221]_y[196]_var[dlwrfsfc].feather.ind
      - 6_drivers/log/feather/NLDAS_time[0.24999]_x[344]_y[132]_var[dlwrfsfc].feather.ind
      - 6_drivers/log/feather/NLDAS_time[0.24999]_x[344]_y[196]_var[dlwrfsfc].feather.ind
```

and the parent remake file:
```yaml
  new_feathers:
    command: loop_tasks(task_plan = new_feather_plan, task_makefile = "6_fetch_new_feather_tasks.yml")
```

Is it better to hard code the .ind file as a target (noting the actual file will change if you change the ind_dir or other naming) or keep as an object?

Reply:
I would hard code. If you keep as an object, and if anything at all depends on that object (including the default target `6_fetch_drivers`, then any work you do on your computer will have to be reproduced on mine, Sam's, Jake's, David's, etc.

Reply:
Each time `create_task_makefile` actually runs, it produces a message telling you the name of that file target you should hard code: https://github.com/USGS-R/scipiper/blob/master/R/create_task_makefile.R#L117. I realize that messages are frequently lost in the remake context, because they only get printed when that particular target actually needs a build, and they can be drowned out by other messages that come later...so if you have any ideas for how better to support this need, please create an issue/PR for it.

Reply:
agreed, but would it be possible to have `loop_tasks` accept an arg that can be used to tell it what indicator to use and what it should be called? Like `target_name` in this case? 
Or is there a reason that the final task plan indicator needs to be in the same directory as the individual task .inds? (I know this is a question for `scipiper` proper, but this thread seems relevant)

Reply:
I didn't see your message 
> Each time create_task_makefile actually runs..
before I commented. I realize that my suggestion wouldn't work because the indicator is specified in `create_task_makefile`, not `loop_tasks`

Reply:
I rather like having loop_tasks always make that final indicator file have the same name as the task yml file - this is one of those cases where I think structure is our friend because it takes away an unnecessary decision, and in a year or two I think we'll all be able to glance at a file named something like `X_phase/log/task_remake_file.ind` and have a really good idea of what it represents. Can you make a case that it's actually a place where we need flexibility?

I'm not following you on the use of `target_name` with `loop_tasks`. Can you write out an example?

You can set a separate value for `ind_dir` in your call to `create_task_plan` (which sets the directory for the task-level .ind files) and for `ind_dir` in your call to `create_task_makefile` (which sets the directory for the final .ind file).

Reply:
Actually, we should be able to have `loop_tasks` at least tell you the name of the final .ind file that is going to be built - could modify this line: https://github.com/USGS-R/scipiper/blob/master/R/loop_tasks.R#L160

Reply:
My use-case was going to be storing big files out of directory of the repository (e.g., on a network drive, or some partition) and keeping the .inds in directory. But now I realize that I can use that pattern now with this workflow w/o any changes needed, since I can specify where the ind directory is independent of the big file dirs.

Reply:
:heavy_check_mark:

Reply:
When the makefile builds from `create_tasl_makefile` - the message that prints out says to run all tasks with:

```6_temp_coop_fetch/log/6_temp_coop_fetch_tasks.ind:```
    ```command: make(I('6_temp_coop_fetch_tasks'), remake_file='6_temp_coop_fetch_tasks.yml')```

Is using `loop_tasks` just a safer way to do this? e.g., provides fault tolerance, retries, etc?

Reply:
Yeah, `loop_tasks` is safer/better. Maybe we should change/remove that suggestion message - I wrote it back before `loop_tasks` was a thing.

Reply:
But when you don't need fault tolerance (like parsing local files), is `loop_tasks` a little overkill? E.g., say I created my makefile and messed up the dir. Instead of getting a quick fail w/ `scmake(I('6_temp_coop_fetch_tasks'), remake_file='6_temp_coop_fetch_tasks.yml'))`, I will get 30+ rapid fire error messages.

Reply:
Yeah, I guess if you specifically want to avoid fault tolerance, and have no interest in ever running just some tasks or some steps in isolation, you shouldn't use loop_tasks. Good point.

If you did want loop_tasks and didn't want 30*n messages, you could set num_tries=1 and task_names=tasks[1] while you're debugging.